3-OO/Ejercicio 1 - POO/ManejoTexto.ts

```diff
@@ -0,0 +1,75 @@
+import * as fs from 'fs';
+​
+export default class ManejoTextos {
+
+    private nombreArchivo : string;
+    private separadorFilas : string;
+    private separadorColumnas : string;
+    private datos : string[][] = [];
+​
+    constructor(archivo: string, separaFil: string, separaCol: string) 
+    {
+        this.nombreArchivo=archivo;
+        this.separadorFilas=separaFil;
+        this.separadorColumnas=separaCol;
+    
+    }
+    
+    public getSeparadorFilas(): string{
+        return this.separadorFilas;
+    }
+
+    public getSeparadorColumnas(): string{
+        return this.separadorColumnas;
+    }
+
+
+    public leerArchivo() 
+    {
+    let fila : string
+    let texto : string = fs.readFileSync(this.nombreArchivo, 'utf8');
+    //separamos las palabras en un arreglo usando el espacio como delimitador
+    console.log(texto);
+    texto.trim();
+    console.log(texto);
+    if (texto) {
+        let filas : string[] = texto.split(this.separadorFilas);
+        let fila : string
+        for (let i = 0; i < filas.length; i++) {
+            fila = filas[i];
+            this.datos.push(fila.split(this.separadorColumnas));                
+        }
+        // filas.forEach(fila => {
+        //     this.datos.push(fila.split(this.separadorColumnas));                
+        // })            
+        }
+    }
+
+    public grabarArchivo(datos : string[]) {
```

Comment:
esto te grabaria solo un registro del archivo. Recorda que datos es string[][]

Reply:
Mauricio antes de llamar a grabar archivo, recorro la matriz y agrego las filas uno detras del otro..
por eso cuando mando a grabar mando solo un String..
for (i = 0; i < info.length; i++) {
    calculadora = info[i];
    salida.push(calculadora.getMarca());
    salida.push(MT.getSeparadorColumnas());
    salida.push(calculadora.getValor().toString());
    salida.push(MT.getSeparadorColumnas());
    bateria = calculadora.getBateria();
    salida.push(bateria.getMarca());
    salida.push(MT.getSeparadorColumnas());
    salida.push(bateria.getTipoBateria());
    salida.push(MT.getSeparadorColumnas());
    bateria.setCarga(14); //modifico la bateria para notar el cambio al grabar archivo..
    salida.push(bateria.getCarga().toString());
    salida.push(MT.getSeparadorFilas());
}